src/core/init/index.js

```diff
@@ -9,7 +9,7 @@ import {isFn} from '../util/core'
 export function initMixin(proto) {
   proto._init = function () {
     const vm = this
-    vm.config = config()
+    vm.config = config(vm)
```

Comment:
Are we sure we should send the whole `vm` ? (same applies for the above)

can we pick some which should be passed and send that only ? I will take a look which I think not needed or good to send . Else we can send the whole for now and change it later if required. 
WDYT ?

Reply:
> can we pick some which should be passed and send that only ?

You mean, like pass certain parts of the API? Not sure.

Plugin authors receive the full `vm` instance in `function(hook, vm) {` right? So maybe it is ok to send the whole thing in the config too.

Another idea could be to make the plugin API more robust. For example, I converted `$docsify` into a function so that I could use `vm` inside the `markdown.renderer.html(html)` hook of the configuration.

But perhaps there should be a way to allow plugins to also modify markdown hooks. In my case, I needed to use `vm.compiler` stuff inside the `html` hook.

Reply:
The `vm` is already public in other parts of the API, so I don't see any reason to not send `vm`. This is covered in #1189